File: javascript/src/models/pollingEndpointMessageOut.ts
// this file is @generated

/** The MessageOut equivalent of polling endpoint */
export interface PollingEndpointMessageOut {
  headers?: { [key: string]: string } | null;
  /** Optional unique identifier for the message */
  eventId?: string | null;
  /** The event type's name */
  eventType: string;
  payload: any;
  /** List of free-form identifiers that endpoints can filter by */
  channels?: string[] | null;
  /** The Message's ID. */
  id: string;
  timestamp: Date;
  tags?: string[] | null;
  deliverAt?: Date | null;
}

export const PollingEndpointMessageOutSerializer = {
  _fromJsonObject(object: any): PollingEndpointMessageOut {
    return {
      headers: object["headers"],
      eventId: object["eventId"],
      eventType: object["eventType"],
      payload: object["payload"],
      channels: object["channels"],
      id: object["id"],
      timestamp: new Date(object["timestamp"]),
      tags: object["tags"],
      deliverAt: object["deliverAt"] ? new Date(object["deliverAt"]) : null,
    };
  },

  _toJsonObject(self: PollingEndpointMessageOut): any {
    return {
      headers: self.headers,
      eventId: self.eventId,
      eventType: self.eventType,
      payload: self.payload,
      channels: self.channels,
      id: self.id,
      timestamp: self.timestamp,
      tags: self.tags,
      deliverAt: self.deliverAt,
    };
  },
};

File: rust/src/models/polling_endpoint_message_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// The MessageOut equivalent of polling endpoint
#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct PollingEndpointMessageOut {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub headers: Option<std::collections::BTreeMap<String, String>>,

    /// Optional unique identifier for the message
    #[serde(rename = "eventId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub event_id: Option<String>,

    /// The event type's name
    #[serde(rename = "eventType")]
    pub event_type: String,

    pub payload: serde_json::Value,

    /// List of free-form identifiers that endpoints can filter by
    #[serde(skip_serializing_if = "Option::is_none")]
    pub channels: Option<std::collections::BTreeSet<String>>,

    /// The Message's ID.
    pub id: String,

    pub timestamp: chrono::DateTime<chrono::Utc>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<std::collections::BTreeSet<String>>,

    #[serde(rename = "deliverAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub deliver_at: Option<chrono::DateTime<chrono::Utc>>,
}

impl PollingEndpointMessageOut {
    pub fn new(
        event_type: String,
        payload: serde_json::Value,
        id: String,
        timestamp: chrono::DateTime<chrono::Utc>,
    ) -> Self {
        Self {
            headers: None,
            event_id: None,
            event_type,
            payload,
            channels: None,
            id,
            timestamp,
            tags: None,
            deliver_at: None,
        }
    }
}
